pmd-java/src/main/java/net/sourceforge/pmd/lang/java/symboltable/ClassScope.java

```diff
@@ -152,10 +152,9 @@ public String getClassName() {
     }
 
     protected Set<NameDeclaration> findVariableHere(JavaNameOccurrence occurrence) {
-        Set<NameDeclaration> result = new HashSet<>();
         Map<MethodNameDeclaration, List<NameOccurrence>> methodDeclarations = getMethodDeclarations();
         Map<VariableNameDeclaration, List<NameOccurrence>> variableDeclarations = getVariableDeclarations();
-        if (occurrence.isThisOrSuper() || occurrence.getImage() != null && occurrence.getImage().equals(className)) {
+        if (occurrence.isThisOrSuper() || className.equals(occurrence.getImage())) {
```

Comment:
True, className is probably not null. You could make "className" a final class member, so that we don't forget to set it, in case we add some time later another contructor... Would it make sense, to do a Null-Check in the constructor?